lib/src/embedded/compiler/message-transformer.test.ts

```diff
@@ -0,0 +1,275 @@
+// Copyright 2020 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import {MessageTransformer} from './message-transformer';
+import {Subject, Observable} from 'rxjs';
+import {take, toArray} from 'rxjs/operators';
+import {InboundMessage, OutboundMessage} from '../../vendor/embedded_sass_pb';
+
+describe('encode', () => {
+  let encoded$: Subject<Buffer>;
+  let transformer: MessageTransformer;
+  let encoded: Promise<Buffer[]>;
+
+  beforeEach(() => {
+    encoded$ = new Subject<Buffer>();
+    transformer = new MessageTransformer(new Observable<Buffer>(), encoded$);
+    encoded = encoded$.pipe(toArray()).pipe(take(1)).toPromise();
+  });
+
+  afterEach(() => transformer.close());
+
+  it('encodes an InboundMessage to buffer', async () => {
+    const input = new InboundMessage.CompileRequest.StringInput();
+    input.setSource('a {b: c}');
+    const request = new InboundMessage.CompileRequest();
+    request.setString(input);
+    request.setId(1);
+    const message = new InboundMessage();
+    message.setCompilerequest(request);
+
+    transformer.write$.next(message);
+    encoded$.complete();
+    expect(await encoded).toEqual([
+      Buffer.from([
+        18,
+        14,
+        8,
+        1,
+        18,
+        10,
+        10,
+        8,
+        97,
+        32,
+        123,
+        98,
+        58,
+        32,
+        99,
+        125,
+      ]),
+    ]);
+  });
+});
+
+describe('decode', () => {
+  let toDecode$: Subject<Buffer>;
+  let transformer: MessageTransformer;
+  let decoded: Promise<OutboundMessage>;
+
+  beforeEach(() => {
+    toDecode$ = new Subject<Buffer>();
+    transformer = new MessageTransformer(toDecode$, new Subject<Buffer>());
+    decoded = transformer.read$
+      .pipe(toArray())
+      .pipe(take(1))
+      .toPromise()
+      .then(messages => messages[0]);
+  });
+
+  afterEach(() => toDecode$.complete());
+
+  it('decodes buffer to OutboundMessage', async () => {
+    toDecode$.next(
+      Buffer.from([
+        18,
+        14,
+        8,
+        1,
+        18,
+        10,
+        10,
+        8,
+        97,
+        32,
+        123,
+        98,
+        58,
+        32,
+        99,
+        125,
+      ])
```

Comment:
Similarly to above, it's probably simpler to just construct a protobuf object and call `serializeBinary()` on it rather than manually encoding the bytes.